### ISSUE TITLE
[eslint-plugin-lit-a11y]: implement label-has-associated-control

### ISSUE THREAD

**Issue by bennypowers** [open-wc/open-wc#1932]
See: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-associated-control.md

# label-has-associated-control

Enforce that a label tag has a text label and an associated control.

There are two supported ways to associate a label with a control:

- Wrapping a control in a label tag.
- Adding the `for` attribute to a label and assigning it a DOM ID that indicates an input within scope of the label.

This rule checks that any `label` tag (or an indicated custom element that implements `label` internally) either
- 1. wraps an `input` element or
- 1. has a `for` attribute and that the `label` tag has text content.

## Rule details

This rule takes one optional object argument of type object:

```json
{
  "rules": {
    "lit-a11y/label-has-associated-control": ["error", {
      "labelCustomElements": [
        "custom-label",
        {
          "tagName": "slotted-label",
          "slot": ["input"]
        }
      ],
    }],
  }
}
```

`labelCustomElements` is a list of custom elements that should be considered labels. If a string is provided, it's assumed that the control is slotted in the anonymous slot. If an object is provided, the control must be slotted into one of the indicated slots.

### Fail
```js
html`
  <label></label>

  <input type="text" />
  <label>Surname</label>
`;
```

### Succeed
```js
html`
  <label for=${for}></label>

  <label for="foo"></label>
  <input id="foo"/>

  <label>
    <input type="text" />
    Surname
  </label>
`;
}
```

## Accessibility guidelines
- [WCAG 1.3.1](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships)
- [WCAG 3.3.2](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions)
- [WCAG 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)

**Comment by bennypowers** [comment]
For this one, we should well-clarify the relationships across shadow boundaries
